### ISSUE TITLE
Purpose of TryAddUpdateKlineCache 

### ISSUE THREAD

**Issue by mysti9uemirage** [glitch100/BinanceDotNet#98]
## Issue Overview
I'm trying to follow and understand https://github.com/glitch100/BinanceDotNet/blob/master/BinanceExchange.Console/ExampleProgram.cs

I have a few questions:
1a. if (!data.Kline.IsBarFinal): What is a final bar? 
1b. What is a keyTime? When I print it out, I see the key time is similar multiple times and then it changes. 
1c. Can you explain what TryAddUpdateKlineCache does in respect to these information?
2a. What is the purpose of the KlineCache in your example? Can I use data.Kline without having it in a cache? 

Please help me understand everything better, especially in regards to the purpose of why stuff are used in the example. Thank you.

**Comment by glitch100** [comment]
This was used as an example simply to show people how to use the API, as per the necessary code from the original spec. I believe (although I don't remember fully) that the `IsBarFinal` represents the last bar in the list provided. So you know where it ends.

**Comment by mysti9uemirage** [comment]
Right, I understand that. I'm trying to understand the example because I figured there's a reason why it was done in such a way and I needed to understand it to follow best practices. 

After changing the code a little bit so that it's simpler and a lot of printouts from ConnectToKlineWebSocket, I finally get that even though the user requests Interval.OneMinute, the data is still streamed (not once per minute) and the start time will be the same for the duration of the interval.  Therefore, now IsBarFinal makes sense because the code will actually have to aggregate all of the data received to make it a minute-bar/ or whatever-interval-bar, else, it'll be a per-event-bar (kline starttime will remain the same for the interval, only the bar.EventTime will change)
